### PR TITLE
Update osx_arm64.txt with rubin_sim

### PR DESCRIPTION
**Pull request by jmeyers314**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)

### CODE REVIEW

File: recipe/migrations/osx_arm64.txt [hunk 1]
```diff
@@ -959,3 +959,4 @@ iteration_utilities
 hmmlearn
 arm_pyart
 stockwell
+rubin_sim
```
Comment (beckermr) [inline recipe/migrations/osx_arm64.txt:962]:
```suggestion
rubin-sim
```

Comment (beckermr) [reply recipe/migrations/osx_arm64.txt:962]: I think it needs to be this?

Comment (jmeyers314) [reply recipe/migrations/osx_arm64.txt:962]: I was following along with https://github.com/lsst/rubin_sim#conda-installation

Comment (beckermr) [reply recipe/migrations/osx_arm64.txt:962]: Right. This is the name of the feedstock, not the package iirc

**Pull request merged**